stripe/src/main/java/com/stripe/android/paymentsheet/CurrencyFormatter.kt

```diff
@@ -7,47 +7,80 @@ import java.util.Locale
 import kotlin.math.pow
 
 internal class CurrencyFormatter {
+
+    fun format(
+        amount: Long,
+        amountCurrencyCode: String,
+        targetLocale: Locale = Locale.getDefault()
+    ) = format(
+        amount,
+        Currency.getInstance(amountCurrencyCode.toUpperCase(Locale.ROOT)),
+        targetLocale
+    )
+
     fun format(
         amount: Long,
-        currency: Currency
+        amountCurrency: Currency,
+        targetLocale: Locale = Locale.getDefault()
     ): String {
-        val currencySymbol = currency.getSymbol(Locale.getDefault())
+        val amountCurrencyDecimalDigits = getDefaultDecimalDigits(amountCurrency)
         val majorUnitAmount =
-            amount / MAJOR_UNIT_BASE.pow(getDefaultFractionDigits(currency).toDouble())
-        val currencyFormat = NumberFormat.getCurrencyInstance()
+            amount / MAJOR_UNIT_BASE.pow(amountCurrencyDecimalDigits.toDouble())
+
+        /**
+         * The currencyFormat for a country and region specifies many things including:
+         * - do they use decimal places (e.g. Korea's is whole numbers only)
+         * - what is the symbol for the currency
+         * - where does the symbol go (right or left of the number)
+         * - how do they format decimal separators (e.g. France uses commas)
+         * - how do they separate thousand digits (e.g. France uses spaces)
+         * When you get the [NumberFormat#getCurrencyInstance] you are getting all this information.
+         *
+         * Some fields not used here, but might be relevant in other scenarios:
+         * - positive and negative numbers
+         *
+         * Among other things the currency format of the amount might have decimal places
+         * even if the targetLocale's currency does not, so below we will do a sort of merge of
+         * the targetLocale and amount currency. We will start with the currency format of the
+         * targetLocale, this sets, most notably, the number decimal and thousands separators,
+         * where the currency symbol should be placed relative to the number. Then we find the
+         * currency symbol of the amount for the targetLocale, and use that. Finally, we set the
+         * number of decimal places used by the amount currency.
+         *
+         * See the [NumberFormat] for why we use a try block.
+         */
+        val currencyFormat = NumberFormat.getCurrencyInstance(targetLocale)
 
-        // We need to cast inside the try catch because most currencies are decimal formats but
-        // not all. See the official Google Docs for NumberFormat for more context.
         runCatching {
```

Comment:
I think an as would work just as well, I will switch it up.   The documentation says a try block for the purpose of a currency that does not return decimal format:

You can also control the display of numbers with such methods as setMinimumFractionDigits. If you want even more control over the format or parsing, or want to give your users more control, you can try casting the NumberFormat you get from the factory methods to a DecimalFormat. This will work for the vast majority of locales; just remember to put it in a try block in case you encounter an unusual one.

Reply:
@mlapadula-stripe's recommendation sounds reasonable to me since it's only there to catch `ClassCastException`